src/sync/models/syncManager.ts

```diff
@@ -25,16 +25,36 @@ export class SyncManager {
     if (syncEntity) {
       throw new SyncAlreadyExistsError(`sync = ${syncEntity.id} already exists`);
     }
+    if (sync.isFull) {
+      const { layerId, geometryType } = sync;
+      const alreadyExistsFullSync = await this.syncRepository.findFullSyncByLayerAndGeometry(layerId, geometryType);
+      if (alreadyExistsFullSync) {
+        throw new FullSyncAlreadyExistsError(
+          `full sync with layer id = ${layerId} and geometry type = ${geometryType} already exists with id ${alreadyExistsFullSync.id}`
+        );
+      }
+    }
     await this.syncRepository.createSync(sync);
   }
 
-  public async updateSync(syncId: string, sync: Omit<Sync, 'id'>): Promise<void> {
+  public async updateSync(syncId: string, updatedSync: Omit<Sync, 'id'>): Promise<void> {
     const syncEntity = await this.syncRepository.findOneSync(syncId);
-    const syncEntityWithId = { id: syncId, ...sync };
 
     if (!syncEntity) {
       throw new SyncNotFoundError(`sync = ${syncId} not found`);
     }
+
+    const { layerId, geometryType } = updatedSync;
+    if (updatedSync.isFull && (layerId != syncEntity.layerId || geometryType != syncEntity.geometryType)) {
+      const alreadyExistsFullSync = await this.syncRepository.findFullSyncByLayerAndGeometry(layerId, geometryType);
+      if (alreadyExistsFullSync) {
+        throw new FullSyncAlreadyExistsError(
+          `full sync with layer id = ${layerId} and geometry type = ${geometryType} already exists with id ${alreadyExistsFullSync.id}`
+        );
+      }
+    }
```

Comment:
maybe its better to prevent changing sync type?

Reply:
done.